docs/references/listener.rst

```diff
@@ -0,0 +1,42 @@
+.. _listener:
+
+Listener
+########
+
+PostgREST uses PostgreSQL `LISTEN <https://www.postgresql.org/docs/current/sql-listen.html>`_ to reload its :ref:`Schema Cache <schema_reloading_notify>` and :ref:`Configuration <config_reloading_notify>` via `NOTIFY <https://www.postgresql.org/docs/current/sql-notify.html>`_.
+This is useful in environments where you can’t send SIGUSR1 or SIGUSR2 Unix Signals.
+Like on cloud managed containers or on Windows systems.
+
+.. code:: postgresql
+
+  NOTIFY pgrst, 'reload schema'; -- reload schema cache
+  NOTIFY pgrst, 'reload config'; -- reload config
+  NOTIFY pgrst;                  -- reload both
+
+By default, the LISTEN channel is enabled (:ref:`db-channel-enabled`) and named ``pgrst`` (:ref:`db-channel`).
+
+Listener on Read Replicas
+=========================
+
+
+The ``LISTEN`` and ``NOTIFY`` commands do not work on PostgreSQL read replicas.
+Thus, if you connect PostgREST to a read replica the Listener will fail to start.
+
+.. code:: postgresql
+
+  NOTIFY pgrst;
+  ERROR:  cannot execute NOTIFY during recovery
```

Comment:
I think we should put the actual PostgREST error here. IIRC it says the same but with additional logs.

Something like this should work:

```
$ postgrest-with-* --replica postgrest-run
```

Maybe you'll have to edit the withTmp on the nix file

Reply:
The error in `psql` is exactly the same:

```
postgres=# notify pgrst;
ERROR:  cannot execute NOTIFY during recovery
```

Now, in the PostgREST 12.2.0 logs, when connecting to a read replica only, it shows:

```
12/Jun/2024:15:19:57 -0500: Failed listening for notifications on the "pgrst" channel. Just "connection to server on socket \"/run/user/1000/postgrest/postgrest-with-postgresql-16-Y3M/socket_replica_6039/.s.PGSQL.5432\" failed: session is read-only\n"
12/Jun/2024:15:19:57 -0500: Failed listening for notifications on the "pgrst" channel. ExitFailure 1
12/Jun/2024:15:19:57 -0500: Attempting to connect to the database...
12/Jun/2024:15:19:57 -0500: Retrying listening for notifications in 8 seconds...
```

Are you referring to the latter?

Reply:
Yeah, maybe just put:

```
12/Jun/2024:15:19:57 -0500: Failed listening for notifications on the "pgrst" channel... connection to server .. failed: session is read-only
```

That error message needs fixing btw. Can be another issue.